server/test/commandHandlers/deployment/ProvideInfrastructureTest.js

```diff
@@ -81,6 +81,35 @@ describe('ProvideInfrastructureCommandHandler:', () => {
     return mod;
   }
 
+  describe('Given a schedule with state of "off"', () => {
+    let promise = null;
+
+    const BLUE_ASG = 'pr1-ta-Web-blue';
+    const GREEN_ASG = 'pr1-ta-Web-green';
+
+    before(() => {
+      let target = createTarget();
+      expectedConfiguration.serverRole.FleetPerSlice = true;
+
+      mocks.infrastructureConfigurationProvider.get
+        .returns(Promise.resolve(expectedConfiguration));
+
+      mocks.autoScalingTemplatesProvider.get
+        .returns(Promise.resolve([{ autoScalingGroupName: GREEN_ASG }, { autoScalingGroupName: BLUE_ASG }]));
+
+      mocks.scheduling.expectedStateFromSchedule = sinon.stub().returns('off');
+
+      promise = target(COMMAND);
+    });
+
+    it('should throw', (done) => {
+      promise.catch((error) => {
+        assert.ok(error);
```

Comment:
Could we assert the type / message of this error to make sure we're actually getting the error we expect, and not something else that's throwing earlier in the chain, eg:

```
assert.equal(error.message.startsWith(`The state of target ASG is currently set to "off"`), true)
```